ipn/prefs.go

```diff
@@ -467,6 +474,106 @@ func (p *Prefs) SetAdvertiseExitNode(runExit bool) {
 		netaddr.IPPrefixFrom(netaddr.IPv6Unspecified(), 0))
 }
 
+// peerWithTailscaleIP returns the peer in st with the provided
+// Tailscale IP.
+func peerWithTailscaleIP(st *ipnstate.Status, ip netaddr.IP) (ps *ipnstate.PeerStatus, ok bool) {
+	for _, ps := range st.Peer {
+		for _, ip2 := range ps.TailscaleIPs {
+			if ip == ip2 {
+				return ps, true
+			}
+		}
+	}
+	return nil, false
+}
+
+func checkIPNotSelf(st *ipnstate.Status, ip netaddr.IP) bool {
```

Comment:
this name isn't clear. boolean funcs with negated names rarely are.

Docs would help, but so would renaming it.

Or just get rid of this and instead make the caller use `!tsaddr.IPsContains(st.TailscaleIPs, ip)` ... but you'd need to add that variant to the `tsaddr` package. We have a dozen variants but not that one.